src/send/SendEnterAmount.tsx

```diff
@@ -0,0 +1,353 @@
+  }, [tokens, defaultTokenIdOverride, lastUsedTokenId])
+
+  // the startPosition and textInputRef variables exist to ensure TextInput
+  // displays the start of the value for long values on Android
+  // https://github.com/facebook/react-native/issues/14845
+  const [startPosition, setStartPosition] = useState<number | undefined>(0)
+  const textInputRef = useRef<RNTextInput | null>(null)
+  const tokenBottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const [token, setToken] = useState<TokenBalance>(defaultToken)
+  const maxAmount = useMaxSendAmount(token.tokenId, FeeType.SEND)
+
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localCurrencyExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const localAmount = useTokenToLocalAmount(new BigNumber(amount), token.tokenId)
+
+  const onTokenPickerSelect = () => {
+    tokenBottomSheetRef.current?.snapToIndex(0)
+    ValoraAnalytics.track(SendEvents.token_dropdown_opened, {
+      currentTokenId: token.tokenId,
+      currentTokenAddress: token.address,
+      currentNetworkId: token.networkId,
+    })
+  }
+
+  const onSelectToken = (token: TokenBalance) => {
+    setToken(token)
+    tokenBottomSheetRef.current?.close()
+    // NOTE: analytics is already fired by the bottom sheet, don't need one here
+  }
+
+  const onMaxAmountPress = () => {
+    setAmount(maxAmount.toString())
+    textInputRef.current?.blur()
+    ValoraAnalytics.track(SendEvents.max_pressed, {
+      tokenId: token.tokenId,
+      tokenAddress: token.address,
+      networkId: token.networkId,
+    })
+  }
+
+  const onReviewPress = () => {
+    // TODO(ACT-943): navigate to send confirmation screen once validation is
+    // done
+    ValoraAnalytics.track(SendEvents.send_amount_continue, {
+      origin,
+      isScan: isFromScan,
+      recipientType: recipient.recipientType,
+      localCurrencyExchangeRate,
+      localCurrency: localCurrencyCode,
+      localCurrencyAmount: localAmount?.toString() ?? null,
+      underlyingTokenAddress: token.address,
+      underlyingTokenSymbol: token.symbol,
+      underlyingAmount: amount.toString(),
+      amountInUsd: null,
+      tokenId: token.tokenId,
+      networkId: token.networkId,
+    })
+  }
+
+  const handleSetStartPosition = (value?: number) => {
+    if (Platform.OS === 'android') {
+      setStartPosition(value)
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader style={{ paddingHorizontal: Spacing.Thick24 }} left={<BackButton />} />
+      <KeyboardAwareScrollView contentContainerStyle={styles.contentContainer}>
+        <Text style={styles.title}>{t('sendEnterAmountScreen.title')}</Text>
+        <View style={styles.inputContainer}>
+          <View style={styles.inputRow}>
+            <TextInput
+              forwardedRef={textInputRef}
+              onChangeText={(value) => {
+                handleSetStartPosition(undefined)
+                setAmount(value)
```

Comment:
This might also require some changes to the clear amount similar to what was done in https://github.com/valora-inc/wallet/blob/main/src/swap/SwapScreen.tsx#L309-L323.

Reply:
ty, updated in 6c8144b